toolkit/livedata/src/main/kotlin/br/com/arch/toolkit/livedata/extention/_liveDataTransformation.kt

```diff
@@ -31,8 +31,16 @@ fun <T, R> LiveData<List<T>>.mapList(transformation: (T) -> R): LiveData<List<R>
 /**
  * Transforms a ResponseLiveData<List<T>> into a ResponseLiveData<List<R>>
  *
+ * @param transformAsync Indicate map will execute synchronously or asynchronously
  * @param transformation Receive the actual non null T value and return the transformed non null R value
  */
+fun <T, R> ResponseLiveData<List<T>>.mapList(
+        transformAsync: Boolean,
+        transformation: (T) -> R
+): ResponseLiveData<List<R>> {
+    return map(transformAsync) { it.map(transformation) }
+}
+
 fun <T, R> ResponseLiveData<List<T>>.mapList(transformation: (T) -> R): ResponseLiveData<List<R>> {
-    return map { it.map(transformation) }
+    return map(false) { it.map(transformation) }
```

Comment:
The mapList method, that is responsible for mapping each item from a list, by default, will not run async or give the developer the possibility to choose between async and sync, I think that would be nice to give the developer this possibility.